core/controllers/profile.py

```diff
@@ -258,26 +258,275 @@ def put(self) -> None:
+                'schema': {
+                    'type': 'dict',
+                    'properties': [{
+                        'name': 'can_receive_email_updates',
+                        'schema': {
+                            'type': 'bool'
+                        }
+                    }, {
+                        'name': 'can_receive_editor_role_email',
+                        'schema': {
+                            'type': 'bool'
+                        }
+                    }, {
+                        'name': 'can_receive_feedback_message_email',
+                        'schema': {
+                            'type': 'bool'
+                        }
+                    }, {
+                        'name': 'can_receive_subscription_email',
+                        'schema': {
+                            'type': 'bool'
+                        }
+                    }]
+                }
+            }
+        }
+    }
 
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
 
     @acl_decorators.can_manage_own_account
     def get(self) -> None:
         """Handles GET requests."""
         assert self.user_id is not None
-        user_settings = user_services.get_user_settings(self.user_id)
         user_email_preferences = user_services.get_email_preferences(
             self.user_id)
+        self.values.update({
+            'can_receive_email_updates': (
+                user_email_preferences.can_receive_email_updates),
+            'can_receive_editor_role_email': (
+                user_email_preferences.can_receive_editor_role_email),
+            'can_receive_feedback_message_email': (
+                user_email_preferences.can_receive_feedback_message_email),
+            'can_receive_subscription_email': (
+                user_email_preferences.can_receive_subscription_email)
+        })
+        self.render_json(self.values)
+
+    @acl_decorators.can_manage_own_account
+    def put(self) -> None:
+        """Handles PUT requests."""
+        assert self.normalized_payload is not None
+        assert self.user_id is not None
+        data = self.normalized_payload['data']
+        bulk_email_signup_message_should_be_shown = (
+            user_services.update_email_preferences(
+                self.user_id, data['can_receive_email_updates'],
+                data['can_receive_editor_role_email'],
+                data['can_receive_feedback_message_email'],
+                data['can_receive_subscription_email']
+            )
+        )
+
+        self.render_json({
+            'bulk_email_signup_message_should_be_shown': (
+                bulk_email_signup_message_should_be_shown)
+        })
+
+
+class ProfilePictureDataUrlHandler(
+    base.BaseHandler[Dict[str, str], Dict[str, str]]
+):
+    """Handles requests for the profile picture data url."""
+
+    URL_PATH_ARGS_SCHEMAS: Dict[str, str] = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {},
+        'PUT': {
+            'data': {
```

Comment:
```suggestion
            'profile_picture_data': {
```

Reply:
Okay, seems like we need to update it in the frontend part as well

Reply:
I have added your suggested  changes but in frontend, as per the lint checks it's in camelCase

Reply:
Makes sense.